dask/dataframe/core.py

```diff
@@ -579,6 +579,17 @@ def _aca_agg(self, token, func, aggfunc=None, **kwargs):
         """ Wrapper for aggregations """
         raise NotImplementedError
 
+    def apply_concat_apply(self, func, aggfunc=None, func_kwargs=None,
+                           aggfunc_kwargs=None):
+        """ Apply a function to each partition, the concat, then apply again
+        """
+        token = '-{}-{}'.format('apply_concat_apply',
```

Comment:
So in map_partitions actually we should probably change it to be something like the following:

```python
    name = '{0}-{1}'.format(name, tokenize(func, metadata, *args, **kwargs))
```